### PR TITLE
feat: add nutanix cluster general info section

### PR DESCRIPTION
**Pull request by radireddy**
ref: MANAGER-7337

Signed-off-by: Ravindra Adireddy <ravindra.adireddy@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `feat/nutanix`
| Bug fix?         | no
| New feature?     | ye
| Breaking change? | no
| Tickets          | Fix #MANAGER-7337
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [ ] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

Implement cluster general info page

### CODE REVIEW

File: packages/manager/modules/manager-components/src/edit-name/template.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<form
+    id="serverEditName"
+    name="serverEditName"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/manager-components/src/edit-name/template.html:3]: Good practice to prefix it with "Form":

```suggestion
    name="serverEditNameForm"
```

File: packages/manager/modules/manager-components/src/edit-name/template.html [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    data-ng-submit="serverEditName.$valid && $ctrl.saveName()"
```
Comment (mohammed-zahaf) [inline packages/manager/modules/manager-components/src/edit-name/template.html:4]: related to this:
https://github.com/ovh/manager/pull/5830/files#r742603788

File: packages/manager/modules/manager-components/src/edit-name/index.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+const moduleName = 'ovhManagerComponentsEditNameComponent';
+
+angular
+  .module(moduleName, [
+    'oui',
+    'pascalprecht.translate',
+    'ngTranslateAsyncLoader',
+  ])
+  .component('editServiceName', component)
```
Comment (mohammed-zahaf) [inline packages/manager/modules/manager-components/src/edit-name/index.js:17]: I think we have to name them like for others module:

```suggestion
const moduleName = 'ovhManagerComponentsEditName';

angular
  .module(moduleName, [
    'oui',
    'pascalprecht.translate',
    'ngTranslateAsyncLoader',
  ])
  .component('editServiceNameComponent', component)
```

File: packages/manager/modules/nutanix/src/dashboard/general-info/edit-display-name/module.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  .module(moduleName, [
+    'oui',
+    'pascalprecht.translate',
+    'ngTranslateAsyncLoader',
+    'ngUiRouterBreadcrumb',
+    'ui.router',
+    editServiceName,
+  ])
+  .config(routing)
+  .component('nutanixGeneralInfoEditDisplayName', component)
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nutanix/src/dashboard/general-info/edit-display-name/module.js:23]: We have to prefix it with 'Component':

```suggestion
  .component('nutanixGeneralInfoEditDisplayNameComponent', component)
```

File: packages/manager/modules/nutanix/src/dashboard/general-info/edit-display-name/routing.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state('nutanix.dashboard.general-info.edit-display-name', {
+    url: '/edit-display-name',
+    component: 'nutanixGeneralInfoEditDisplayName',
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nutanix/src/dashboard/general-info/edit-display-name/routing.js:4]: Related to this:
https://github.com/ovh/manager/pull/5830/files#r742617966

File: packages/manager/modules/nutanix/src/dashboard/general-info/module.js [hunk 6]
```diff
@@ -17,6 +19,8 @@ angular
     'ui.router',
     ovhManagerBillingComponents,
     serverSupport,
+    region,
+    editName,
   ])
   .config(routing)
   .component('nutanixGeneralInfo', component)
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nutanix/src/dashboard/general-info/module.js:26]:
```suggestion
  .component('nutanixGeneralInfoComponent', component)
```

File: packages/manager/modules/nutanix/src/dashboard/general-info/routing.js [hunk 7]
```diff
@@ -3,8 +3,6 @@ export default /* @ngInject */ ($stateProvider) => {
     url: '',
     component: 'nutanixGeneralInfo',
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nutanix/src/dashboard/general-info/routing.js:4]: related to this:
https://github.com/ovh/manager/pull/5830/files#r742619275

File: packages/manager/modules/nutanix/src/dashboard/nodes/module.js [hunk 8]
```diff
@@ -17,6 +19,7 @@ angular
     'pascalprecht.translate',
     'ui.router',
     node,
+    region,
   ])
   .config(routing)
   .component('nutanixNodes', component)
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nutanix/src/dashboard/nodes/module.js:25]:
```suggestion
  .component('nutanixNodesComponent', component)
```

File: packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/routing.js [hunk 9]
```diff
@@ -51,9 +51,9 @@ export default /* @ngInject */ ($stateProvider) => {
         }),
       goToNutanixNode: /* @ngInject */ ($state, Alerter) => (
         message = false,
-        type = 'success',
+        type = 'DONE',
       ) => {
-        const reload = message && type === 'success';
+        const reload = message && type === 'DONE';
```
Comment (mohammed-zahaf) [inline packages/manager/modules/nutanix/src/dashboard/nodes/node/general-info/routing.js:56]: May be could be part of const file or variable ?

File: packages/manager/modules/manager-components/src/edit-name/index.js (hunk 3 above)
Comment (radireddy) [reply packages/manager/modules/manager-components/src/edit-name/index.js:17]: 👍

File: packages/manager/modules/nutanix/src/dashboard/general-info/module.js (hunk 6 above)
Comment (radireddy) [reply packages/manager/modules/nutanix/src/dashboard/general-info/module.js:26]: Change not related to this PR. Need to do a refactoring to make this change. It need to be done as part of a new PR.

File: packages/manager/modules/nutanix/src/dashboard/general-info/routing.js (hunk 7 above)
Comment (radireddy) [reply packages/manager/modules/nutanix/src/dashboard/general-info/routing.js:4]: Change not related to this PR. Need to do a refactoring to make this change. It need to be done as part of a new PR.

File: packages/manager/modules/nutanix/src/dashboard/nodes/module.js (hunk 8 above)
Comment (radireddy) [reply packages/manager/modules/nutanix/src/dashboard/nodes/module.js:25]: Change not related to this PR. Need to do a refactoring to make this change. It need to be done as part of a new PR.

File: packages/manager/modules/nutanix/src/dashboard/general-info/template.html [hunk 10]
```diff
@@ -1,4 +1,139 @@
...
+            </oui-tile-definition>
+
+            <!-- deployment mode -->
+            <oui-tile-definition
+                term="{{:: 'nutanix_dashboard_cluster_deployment_mode' | translate }}"
+            >
+                <oui-tile-description>
+                    <div class="oui-tile__description">
+                        <div
+                            data-ng-if="$ctrl.cluster.targetSpec.rackAwareness"
```
Comment (varun257) [inline packages/manager/modules/nutanix/src/dashboard/general-info/template.html:73]: I think we can add `getter` in `Cluster` class to fetch these values.

File: packages/manager/modules/nutanix/src/dashboard/general-info/routing.js [hunk 11]
```diff
@@ -15,6 +13,32 @@ export default /* @ngInject */ ($stateProvider) => {
...
+      goToEditName: /* @ngInject */ ($state) => (displayName) =>
+        $state.go('nutanix.dashboard.general-info.edit-display-name', {
+          displayName,
+        }),
+      goToNutanixGeneralInfo: /* @ngInject */ (
+        $state,
+        Alerter,
+        serviceName,
+      ) => (message = false, type = 'DONE') => {
+        const reload = message && type === 'DONE';
```
Comment (varun257) [inline packages/manager/modules/nutanix/src/dashboard/general-info/routing.js:25]: `'DONE'` can be put in constant as done in other place as `STATUS_DONE`

**Review by mohammed-zahaf — APPROVED** [review]

**Review by varun257 — APPROVED** [review]

**Pull request merged**